### DISCUSSION TITLE
FastApi is not returning "extra" fields of a model

### DISCUSSION THREAD

**Discussion by JobaDiniz** [fastapi/fastapi#11896] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class NocodeuiBaseModel(BaseModel):
    """
    A base pydantic model class that sets standard configuration for serialization and deserialization.
    """

    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)


class NocodeuiResponse(NocodeuiBaseModel, Generic[TResponse]):
    links: dict[str, Link] = Field(..., alias="_links")
    data: TResponse

    model_config = ConfigDict(extra="allow")


@router.get("/{id}/parameters", name="get_flow_parameters", response_model=NocodeuiResponse[Sequence[FlowParameter]])
@inject
@cache(expire=30)
async def get_flow_parameters(
    request: Request, id: str, flow_repository: FlowRepository = Depends(Provide[Container.repository])
):
    parameters = await flow_repository.get_parameters(id=id)
    # TODO: this "flow" shouldn't be returned but the research code needs. this can be changed in the UI later
    flow = await flow_repository.get_flow(id=id)
    return NocodeuiResponse(
        links=LinkBuilder.build_self_link(request, "get_flow", id=id),
        data=parameters,
        flow=_create_flow_model_response(request, flow),
    )
```

### Description

I'm adding an extra "flow" field to the `NocodeuiResponse`, which is configured to allow it.

When returning this pydantic model, fastapi is not including the "flow" field in the JSON response. It's being ignored.

How can I fix that?

### Operating System

Windows

### FastAPI Version

0.111.1

### Pydantic Version

2.6.4

### Python Version

Python 3.11.4

**Comment by YuriiMotov** [comment, marked as answer]
This is the feature of FastAPI. This way it ensures that you return only particular part of data (specified in the response model).
This helps you prevent data leaking.

If you want to bypass it, you can return `JSONResponse` from your endpoint. This way FastAPI will not validate your response and send all fields including extra fields.

```
    resp = NocodeuiResponse(
        links=LinkBuilder.build_self_link(request, "get_flow", id=id),
        data=parameters,
        flow=_create_flow_model_response(request, flow),
    )
    return JSONResponse(content=resp.model_dump_json(exclude={"flow"}))
```

The code above will return all fields except `flow`.

But remember: when you return `JSONResponse` directly, response validation is your responsibility!
